Tella/Data/Database/Vault/VaultDataBase.swift

```diff
@@ -0,0 +1,352 @@
+//
+//  Copyright © 2023 HORIZONTAL. All rights reserved.
+//
+
+import Foundation
+
+
+protocol VaultDataBaseProtocol {
+    func createVaultTable()
+    func addVaultFile(file : VaultFileDB, parentId: String?) throws
+    func getVaultFiles(parentId: String?, filter: FilterType?, sort: FileSortOptions?) -> [VaultFileDB]
+    func getVaultFile(id: String?) -> VaultFileDB?
+    func getVaultFiles(ids: [String]) -> [VaultFileDB]
+    func getRecentVaultFiles() -> [VaultFileDB]
+    func renameVaultFile(id: String?, name: String?)
+    func moveVaultFile(fileIds: [String], newParentId: String?)
+    func deleteVaultFile(ids: [String])
+}
+
+protocol DataBase {
+    var dataBaseHelper : DataBaseHelper { get }
+    var statementBuilder : SQLiteStatementBuilder { get  }
+    func checkVersions()
+    func createTables()
+}
+
+class VaultDatabase : DataBase, VaultDataBaseProtocol {
+    
+    var dataBaseHelper: DataBaseHelper
+    var statementBuilder: SQLiteStatementBuilder
+    
+    private var rootId = "11223344-5566-4777-8899-aabbccddeeff";
+    
+    init(key: String?) {
+        dataBaseHelper = DataBaseHelper(key: key, databaseName: VaultD.databaseName)
```

Comment:
Instead of creating `DataBaseHelper` and `SQLiteStatementBuilder` instances directly within the `VaultDatabase` class, consider injecting them as dependencies.